### ISSUE TITLE
I'd love to be able to read from s3, like in Apache Arrow

### ISSUE THREAD

**Issue by isaacsanders** [elixir-nx/explorer#126]

**Comment by kimjoaoun** [comment]
You mean reading a dataframe using something like this example below? 

```
Explorer.DataFrame.from_bucket("arn:aws:s3:::bucket_name/key_name")
```

**Comment by isaacsanders** [comment]
Or via URL, `s3://bucket-name/data/date=2022-03-01/`

**Comment by isaacsanders** [comment]
Arrow defines “file system” types that enable this behavior.

**Comment by kimjoaoun** [comment]
@isaacsanders wouldn't this require some sort of authentication with the AWS IAM?

**Comment by philss** [comment]
I like the idea but I would suggest to not include such feature in the project right now because of the dependencies we would need to require.
Maybe a good approach would be to create such functionality in a separated project first. WDYT?

**Comment by philss** [comment]
Yeah, I changed my mind. I think it worth to add this feature :+1:

**Comment by cigrainger** [comment]
I agree. I think the example [here](https://github.com/jorgecarleitao/arrow2/tree/b70483ed6a56760d6ad514841a613885ca4a9eeb/examples/s3) should give us a good start. I'll look into this.

**Comment by josevalim** [comment]
This will be closed as part of #190.
